corsx/middleware.go

```diff
@@ -19,6 +19,8 @@ import (
 //	  panic("implement me")
 //	})
 //	// ...
+//
+// Deprecated: because this is not really practical to use, you should use CheckOrigin as the cors.Options.AllowOriginRequestFunc instead.
```

Comment:
The problem with this was that the config has to come from the config provider, but where you initialize the middleware you have no access to the config.